packages/core/src/crawlers/error_snapshotter.ts

```diff
@@ -0,0 +1,135 @@
+import crypto from 'node:crypto';
+
+import type { ErrnoException } from './error_tracker';
+import type { CrawlingContext } from '../crawlers/crawler_commons';
+import type { KeyValueStore } from '../storages';
+import type { PlaywrightCrawlingContext, PuppeteerCrawlingContext, PlaywrightPage, PuppeteerPage } from '../typedefs';
+
+const { PWD, CRAWLEE_STORAGE_DIR, APIFY_IS_AT_HOME } = process.env;
+
+/**
+ * ErrorSnapshotter class is used to capture a screenshot of the page and a snapshot of the HTML when an error occur during web crawling.
+ */
+export class ErrorSnapshotter {
+    static readonly MAX_ERROR_CHARACTERS = 30;
+    static readonly MAX_HASH_LENGTH = 30;
+    static readonly MAX_FILENAME_LENGTH = 250;
+    static readonly BASE_MESSAGE = 'An error occurred';
+    static readonly SNAPSHOT_PREFIX = 'ERROR_SNAPSHOT';
+    static readonly KEY_VALUE_PLATFORM_PATH = 'https://api.apify.com/v2/key-value-stores';
+    static readonly KEY_VALUE_STORE_LOCAL_PATH = `file://${PWD}/${CRAWLEE_STORAGE_DIR || 'storage'}/key_value_stores`;
```

Comment:
you shouldn't work with the evn vars directly, at least no with those that are handled via `Configuration` class, some values could be specified only there instead of actual env vars (e.g. when using `crawlee.json` config file).